adam/perception/perception_graph.py

```diff
@@ -561,6 +567,27 @@ def matches(
             debug_mapping_sink.clear()
             debug_mapping_sink.update(matching.debug_largest_match)
 
+    def rendering_debug_function(
```

Comment:
Pull the debug function and its counter into its own object (which implements `__call__`)